core/src/main/java/org/apache/calcite/rel/logical/LogicalProject.java

```diff
@@ -123,12 +143,27 @@ public static LogicalProject create(final RelNode input, List<RelHint> hints,
         cluster.traitSet().replace(Convention.NONE)
             .replaceIfs(RelCollationTraitDef.INSTANCE,
                 () -> RelMdCollation.project(mq, input, projects));
-    return new LogicalProject(cluster, traitSet, hints, input, projects, rowType);
+    return new LogicalProject(cluster, traitSet, hints, input, projects, rowType,
+        ImmutableSet.of());
+  }
```

Comment:
Could you delegate to the new method instead of duplicating the logic?

Reply:
fixed